### PR TITLE
Upgrade Polygon Oracle Tunnel

### PR DESCRIPTION
**Pull request by nicholaspai**

### CODE REVIEW

File: UMIPs/umip-113.md [hunk 1]
```diff
@@ -10,7 +10,7 @@
 This UMIP will have the effect of registering an oracle root tunnel contract that will enable the optimistic oracle deployed on Polygon to make cross-chain price requests to the DVM on mainnet Ethereum.
 
 ## Motivation
-[This document](https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/polygon/README.md) describes a two way tunnel architecture that is built on top of the [canonical state transfer mechanism](https://docs.matic.network/docs/develop/l1-l2-communication/state-transfer) between Polygon and Ethereum. Data is sent between the networks through the tunnel using the same public infrastructure that you use to deposit ERC20 tokens from Ethereum to Polygon and vice versa. The only trusted third parties are the Polygon validators.
```
Comment (nicholaspai) [inline UMIPs/umip-113.md:13]: Upgrading links in UMIP-113

File: UMIPs/umip-placeholder-root-tunnel.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+| Status     | Draft                                                                                                                                    |
+| Created    | March 28, 2022                                                                                                                           |
+
+## Summary (2-5 sentences)
+This UMIP will have the effect of upgrading the oracle root tunnel contract that will enable the optimistic oracle deployed on Polygon to make cross-chain price requests to the DVM on mainnet Ethereum.
+
+## Motivation
+The utility of the oracle root tunnel system is described in [UMIP-113](https://github.com/UMAprotocol/UMIPs/blob/master/UMIPs/umip-113.md) and the explanation still holds true.
+
+The motivation for upgrading the `OracleRootTunnel` is out of necessity. Polygon changed the `FxBaseRootTunnel` implementation in September 2021 to be able to accept proofs generated on type 1 (i.e. EIP1559/London) transactions submitted on the Polygon network.
```
Comment (mrice32) [inline UMIPs/umip-placeholder-root-tunnel.md:15]: nit: I thought EIP1559 transactions were type 2? https://eips.ethereum.org/EIPS/eip-1559

**Review by mrice32 — APPROVED** [review]

File: UMIPs/umip-placeholder-child-tunnel.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+## Technical Specification
+To accomplish this upgrade, the following actions need to be taken:
+- The newly deployed `OracleChildTunnel` will need to be set as the "Oracle" in hex (i.e. `0x4f7261636c65`) in the Polygon `Finder`. The specific function data that needs to be sent to Polygon is `polygonFinder.changeImplementationAddress(0x4f7261636c65, newOracleChildTunnelAddress)` so that the `OptimisticOracle` on Polygon can fetch price resolution data through the new child tunnel.
+
+## Rationale
+This UMIP will enable type 1 transactions on Polygon to be able to relay messages to Ethereum, which is critical for Polygon price requests to be able to be secured by the Ethereum DVM.
+
+## Implementation
+
+The changes to the root tunnel contract can be found [here](https://github.com/UMAprotocol/protocol/pull/3863) and they have not been audited. The child tunnel is not changed.
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply UMIPs/umip-placeholder-child-tunnel.md:28]: No IDTS

File: UMIPs/umip-placeholder-root-tunnel.md (hunk 2 above)
Comment (nicholaspai) [reply UMIPs/umip-placeholder-root-tunnel.md:15]: oh youre right

File: UMIPs/umip-placeholder-child-tunnel.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
+## Headers
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply UMIPs/umip-placeholder-child-tunnel.md:1]: Well the main hiccup is that we have two separate admin scripts `changeFinderAddress` and `registerContract` that propose 2 admin transactions, but for this UMIP we can write a 1-off script that bundles these two actions into one proposal.

[the remark this answers is not in the document]
Comment (nicholaspai) [reply UMIPs/umip-placeholder-child-tunnel.md:1]: Added a script in this PR: https://github.com/UMAprotocol/protocol/pull/3868

Will consolidate these UMIPs now

File: UMIPs/umip-placeholder-child-tunnel.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (nicholaspai) [reply UMIPs/umip-placeholder-child-tunnel.md]: Actually looks like it has an audit but the link is broken: https://github.com/fx-portal/contracts/tree/baed24d22178201bca33140c303e0925661ec0ac#fx-portalflexible-portal

**Review by mrice32 — APPROVED** [review]
LGTM!

**Review by chrismaree — APPROVED** [review]

**Review by therealhexi — APPROVED** [review]
Nice!

**Pull request merged**